src/shared/lib/MutationTableDownloadDataFetcher.ts

```diff
@@ -86,6 +94,17 @@ export class MutationTableDownloadDataFetcher implements ILazyMobXTableApplicati
         }
     }
 
+    private async fetchAllGenomeNexusFunctionalImpactData()
+    {
+        if (this.mutationData.result)
+        {
+            return await fetchGenomeNexusFunctionalImpactData(this.mutationData.result);
```

Comment:
since the function that calls this is going to need to handle the response as a promise (await or .then) there is no reason to put await here.  you can just do return fetchGenomeNexusFunc(...)
Then you can take the async tag off.
In the else condition, you will need to return Promise.resolve(undefined).